client/app/reducers/selection.coffee

```diff
@@ -0,0 +1,38 @@
+Immutable = require 'immutable'
+
+{ActionTypes} = require '../constants/app_constants'
+
+DEFAULTSTATE =
+    selected: Immutable.Set()
+    allSelected: false
+
+module.exports = (state = DEFAULTSTATE, action) ->
+    switch action.type
+        when ActionTypes.MAILBOX_SELECT_ALL
+            if state.allSelected
+                nextState = DEFAULTSTATE
```

Comment:
What about having a dedicated action `ActionTypes.MAILBOX_DESELECT_ALL`. It seems that this part is too much related to a checkbox UI action. If we consider only this checkbox, it makes sense to select/unselect all on every check, i.e. dispatch a `SELECT_ALL` action. But maybe at